### PR TITLE
LL-6359 - SWAP - Target account row

### PR DESCRIPTION
**Pull request by jgiamporcaro-ledger**
⚠️ This PR includes the commits from #4142, that is why the PR is still in draft.

Synchronise the target account row with the swap state.
In addition to that, I removed some data we used until now to ensure that everything was designed and polished some part of the UI.

## 🦒 Context (issues, jira)

[LL-6359]

## 💻  Description / Demo (image or video)

<details>

<summary>Screen recording</summary>

https://user-images.githubusercontent.com/33158502/131002721-97551896-dc8a-43c0-a54d-e49ec4e659f2.mov

</details>

## 🖤  Expectations to reach

PR must pass CI, merge develop if conflicts, do not force push. Thanks!

- **on QA**: at least one of these two checkboxes must be checked:
  - [ ] a specific test planned is defined on Jira
  - [ ] this PR is covered by automatic UI test
- **on delivery**: at least one of these two checkboxes must be checked: 
  - [x] Option 1: **no impact**: The changes of this PR have ZERO impact on the userland (invisible for users)
  - [ ] Option 2: **atomic delivery**: the changes is atomic and complete (no partial delivery)

PR must pass CI, merge develop if conflicts, do not force push. Thanks!

[LL-6359]: https://ledgerhq.atlassian.net/browse/LL-6359

### CODE REVIEW

File: src/renderer/screens/exchange/Swap2/utils/shared/useSwapTransaction.js [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
 import { shallowAccountsSelector } from "~/renderer/reducers/accounts";
 
-const useSwapTransaction = () => {
+export type SwapSelectorStateType = {
+  currency: null | TokenCurrency | CryptoCurrency,
+  account: null | Account | TokenAccount,
+  parentAccount: null | Account,
+  amount: null | BigNumber,
+};
+const SelectorStateDefaultValues = {
```
Comment (juan-cortes) [inline src/renderer/screens/exchange/Swap2/utils/shared/useSwapTransaction.js:26]: camelcase?

File: src/renderer/screens/exchange/Swap2/Form/FormSummary/SectionTarget.js [hunk 2]
```diff
@@ -4,91 +4,116 @@ import styled from "styled-components";
...
   column-gap: 0.75rem;
   padding: 0.75rem;
   border-radius: 4px;
+  align-items: center;
 `;
 
 const Text: ThemedComponent<{}> = styled(TextBase).attrs(() => ({
   ff: "Inter",
-  fontSize: "12px",
+  fontSize: "0.875rem",
```
Comment (juan-cortes) [inline src/renderer/screens/exchange/Swap2/Form/FormSummary/SectionTarget.js:33]: :wat:

**Review by juan-cortes — APPROVED** [review]
it sucks that we have duplicated commits in all prs, it's hard to review like this.

**Comment by jelbaz-ledger** [comment]
Merging since this PR is blocking #4150 and nothing is critical in the code review 📝 (it can be fixed directly in the feature branch)

**Pull request merged**